packages/runtime/src/tutorial-runner.ts

```diff
@@ -302,6 +302,43 @@ export class TutorialRunner {
     );
   }
 
+  /**
+   * Get snapshot of runner's current files.
+   * Also prepares `package.json`'s `stackblitz.startCommand` with runner's commands.
+   *
+   * Note that file paths do not contain the leading `/`.
+   */
+  takeSnapshot() {
+    const files: Record<string, string> = {};
+
+    // first add template files
+    for (const [filePath, value] of Object.entries(this._currentTemplate || {})) {
+      if (typeof value === 'string') {
+        files[removeLeadingSlash(filePath)] = value;
+      }
+    }
+
+    // next overwrite with files from editor
+    for (const [filePath, value] of Object.entries(this._currentFiles || {})) {
+      if (typeof value === 'string') {
+        files[removeLeadingSlash(filePath)] = value;
+      }
+    }
+
+    const packageJson = parseJson(files['package.json']);
```

Comment:
We should probably reuse `this._packageJsonContent`? I just realised that the `package.json`, if present, might be in a sub folder.

Note that the path, if it's at the root, will be `/package.json` instead of `package.json`.

Reply:
The `_packageJsonContent` doesn't contain contents of `package.json` if that file is not present in `_files`. We want to read `package.json`'s from templates too. 

Should I make changes to pass `{...files, ...template}` here?

https://github.com/stackblitz/tutorialkit/blob/298615748b1f2d3ea737c591ce193eb0d28407ca/packages/runtime/src/tutorial-runner.ts#L191

It's then set here:

https://github.com/stackblitz/tutorialkit/blob/298615748b1f2d3ea737c591ce193eb0d28407ca/packages/runtime/src/tutorial-runner.ts#L416-L420

Reply:
Oh good catch! That's indeed a bug! Yep, line 191 should definitely see `{...files, ...template}`.

Reply:
We also need to do add a:

```
this._packageJsonPath = filePath;
```

To know where the file is and reuse that when modifying it.